frontend/src/components/modals/settings/SettingsModal.tsx

```diff
@@ -25,6 +25,9 @@ function SettingsModal({ isOpen, onOpenChange }: SettingsProps) {
   const [agents, setAgents] = React.useState<string[]>([]);
   const [settings, setSettings] =
     React.useState<Partial<Settings>>(currentSettings);
+  const [apiKey, setApiKey] = React.useState<string>(
```

Comment:
You've extended `getCurrentSettings` to also include the API key. Maybe you can utilize the `settings` state that already stores this data directly rather than use a separate state to handle the API key.

Reply:
Yep, good point.